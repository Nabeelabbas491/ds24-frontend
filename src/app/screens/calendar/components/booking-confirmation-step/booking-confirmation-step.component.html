<ds-booking-confirmation></ds-booking-confirmation>
