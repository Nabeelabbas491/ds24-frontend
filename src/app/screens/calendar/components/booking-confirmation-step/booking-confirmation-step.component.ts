import { Component } from '@angular/core';

@Component({
  selector: 'ds-booking-confirmation-step',
  templateUrl: './booking-confirmation-step.component.html',
})
export class BookingConfirmationStepComponent {}
