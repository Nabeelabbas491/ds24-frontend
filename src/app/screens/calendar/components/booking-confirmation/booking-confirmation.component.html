<div
  class="bg-neutral-n0 w-100 my-0 mx-auto flex flex-col justify-start items-center py-5 px-10"
  *ngIf="bookingSuccess$ | async as bookingSuccess"
>
  <ds-icon
    [name]="icons.Checkmark"
    [color]="iconColors.Success300"
    [size]="48"
    [title]="'CALENDAR.PREVIOUS_MONTH' | translate"
  ></ds-icon>
  <h3 class="text-neutral-n500 font-medium mt-5 break-words w-full text-center">
    {{ 'BOOKING.BOOKING_CONFIRMATION.BOOKING_NAME' | translate: { bookingName: bookingSuccess.name } }},
    {{ 'BOOKING.BOOKING_CONFIRMATION.BOOKING_CONFIRMED' | translate }}
  </h3>
  <h4 class="font-medium mt-5">
    {{ bookingSuccess.dayTitle | date: 'fullDate' }} - {{ bookingSuccess.timeSlot | timeSlot }}
  </h4>
  <p class="text-neutral-n500 mt-5">{{ 'BOOKING.BOOKING_CONFIRMATION.CALENDAR_INVITATION_SENT' | translate }}</p>
  <a class="cursor-pointer underline mt-3 text-base" href="#">{{
    'BOOKING.BOOKING_CONFIRMATION.SEND_CONFIRMATION_EMAIL_AGAIN' | translate
  }}</a>
  <div class="my-5" *ngIf="bookingSuccess.meetingTypeDetail?.id === meetingTypeIds.Zoom">
    <h4 class="font-medium my-3 text-center">{{ 'BOOKING.BOOKING_CONFIRMATION.LINK_TO_ZOOM_MEETING' | translate }}</h4>
    <ds-copy-info-box
      [copyText]="bookingSuccess.zoomLink"
      (click)="copyZoomLink(bookingSuccess.zoomLink)"
    ></ds-copy-info-box>
  </div>
  <div class="my-5" *ngIf="bookingSuccess.meetingTypeDetail?.id === meetingTypeIds.Inbound">
    <h4 class="font-medium my-3 text-center">{{ 'BOOKING.BOOKING_CONFIRMATION.VENDOR_PHONE_NUMBER' | translate }}</h4>
    <ds-copy-info-box
      [copyText]="bookingSuccess.vendorPhoneNumber"
      (click)="copyPhone(bookingSuccess.vendorPhoneNumber)"
    ></ds-copy-info-box>
  </div>
  <div class="mt-3 flex justify-start flex-row mobile-view:flex-col">
    <button dsButton="default" class="mr-2 mobile-view:w-full" (click)="downloadICS(bookingSuccess.appointmentId)">
      {{ 'BOOKING.BOOKING_CONFIRMATION.DONWNLOAD_ICS' | translate }}
    </button>
    <button
      dsButton="default"
      class="mobile-view:w-full mobile-view:mt-3"
      [disabled]="(isBookingSynced$ | async) === true"
      (click)="saveToGoogleCalendar()"
    >
      {{ 'BOOKING.BOOKING_CONFIRMATION.SAVE_TO_GOOGLE_CALENDAR' | translate }}
    </button>
  </div>
  <div class="flex flex-row justify-center mt-3">
    <ds-spinner
      [color]="spinnerColor"
      class="mx-auto"
      *ngIf="(downloadClientICSPending$ | async) || (getAuthenticationUrlPending$ | async)"
      [size]="30"
    ></ds-spinner>
  </div>
</div>
