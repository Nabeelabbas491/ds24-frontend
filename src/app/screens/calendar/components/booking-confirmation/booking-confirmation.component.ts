import { Component } from '@angular/core';
import { IconColor, IconName } from '@ds24/elements';
import { actions, selectors } from '../../../../store';
import { Store } from '@ngrx/store';
import { MeetingTypeId } from './../../../../types/booking.type';
import { Clipboard } from '@angular/cdk/clipboard';
import { TranslateService } from '@ngx-translate/core';
import { ReactiveLifecycles } from '@ds24/utilities';
import { GoogleIntegrationInfo } from './../../../../types/google-integration.type';
import { takeUntil } from 'rxjs';
import { saveBookingSnapshot } from './../../../../shared/common/util';

@Component({
  selector: 'ds-booking-confirmation',
  templateUrl: './booking-confirmation.component.html',
})
export class BookingConfirmationComponent extends ReactiveLifecycles {
  meetingTypeIds = MeetingTypeId;

  icons: typeof IconName = IconName;
  iconColors: typeof IconColor = IconColor;
  spinnerColor = IconColor.Primary300;

  bookingSuccess$ = this.store.select(selectors.booking.selectBookingSaveSuccess);
  downloadClientICSPending$ = this.store.select(selectors.bookingAPI.selectDownloadClientICSPending);
  bookingSnapshot$ = this.store.select(selectors.booking.selectBookingSnapshot);
  isBookingSynced$ = this.store.select(selectors.booking.selectIsBookingSynced);

  bookingProductId$ = this.store.select(selectors.bookingAPI.selectBookingProductId);
  bookingProductId: string | null | undefined;

  getAuthenticationUrlPending$ = this.store.select(selectors.googleIntegration.selectAuthenticationUrlPending);

  constructor(
    private store: Store,
    private clipboard: Clipboard,
    private _translate: TranslateService,
  ) {
    super();

    this.bookingProductId$.pipe(takeUntil(this.ngOnDestroy$)).subscribe(bookingProductId => {
      this.bookingProductId = bookingProductId;
    });
  }

  downloadICS(appointmentId: number) {
    this.store.dispatch(actions.bookingAPI.downloadClientICS({ appointmentId }));
  }
  saveToGoogleCalendar() {
    if (this.bookingProductId) {
      this.bookingSnapshot$.pipe(takeUntil(this.ngOnDestroy$)).subscribe(bookingSnapshot => {
        saveBookingSnapshot(bookingSnapshot);

        const integrationInfo: GoogleIntegrationInfo = {
          integrationMode: 'client-booking',
          bookingProductId: this.bookingProductId!,
        };

        this.store.dispatch(actions.googleIntegration.getAuthenticationUrl({ integrationInfo }));
      });
    }
  }

  copyPhone(phone: string) {
    this.clipboard.copy(phone);
    this.showSnackbarMessage('BOOKING.BOOKING_CONFIRMATION.VENDOR_PHONE_NUMBER_COPIED');
  }
  copyZoomLink(zoomLink: string) {
    this.clipboard.copy(zoomLink);
    this.showSnackbarMessage('BOOKING.BOOKING_CONFIRMATION.ZOOME_MEETING_LINK_COPIED');
  }

  showSnackbarMessage(message: string) {
    const snackBarMessage = this._translate.instant(message);
    this.store.dispatch(actions.notification.callSnackBarPopup({ snackBarMessage }));
  }
}
