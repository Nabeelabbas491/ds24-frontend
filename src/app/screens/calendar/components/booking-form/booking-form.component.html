<div>
  <h4 class="font-medium">{{ 'BOOKING.BOOKING_FORM.ENTER_DETAILS' | translate }}</h4>
  <form [formGroup]="form" novalidate (submit)="confirmBooking()" #bookingForm="ngForm">
    <ds-input-wrapper class="mt-2 text-left">
      <input dsInput formControlName="name" type="text" [placeholder]="'BOOKING.BOOKING_FORM.NAME' | translate" />
      <ds-form-label>{{ 'BOOKING.BOOKING_FORM.NAME' | translate }}</ds-form-label>
    </ds-input-wrapper>
    <ds-form-sub-text *ngIf="leftCountName$ | async as leftCountName" class="text-left">
      {{ 'BOOKING.BOOKING_FORM.NAME_DESCRIPTION' | translate: { value: leftCountName } }}
    </ds-form-sub-text>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.name?.hasError('required') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.NAME_REQUIRED' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.name?.hasError('maxlength') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.NAME_MAX_LENGTH' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.name?.hasError('whitespace') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.ENTER_VALID_TEXT' | translate
      }}</ds-form-error>
    </div>
    <ds-input-wrapper class="mt-2 text-left">
      <input
        dsInput
        formControlName="email"
        type="email"
        [placeholder]="'BOOKING.BOOKING_FORM.EMAIL' | translate"
        email
      />
      <ds-form-label>{{ 'BOOKING.BOOKING_FORM.EMAIL' | translate }}</ds-form-label>
    </ds-input-wrapper>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.email?.hasError('required') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.EMAIL_REQUIRED' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.email?.hasError('maxlength') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.EMAIL_MAX_LENGTH' | translate
      }}</ds-form-error>
      <ds-form-error
        *ngIf="
          !formControls.email?.hasError('maxlength') && formControls.email?.hasError('email') && bookingForm.submitted
        "
        >{{ 'BOOKING.BOOKING_FORM.EMAIL_NOT_IN_VALID_FORMAT' | translate }}</ds-form-error
      >
    </div>
    <ng-container *ngIf="meetingType$ | async as meetingType">
      <ds-input-wrapper class="mt-2 text-left" *ngIf="meetingType.id === meetingTypeIds.Outbound">
        <input
          dsInput
          formControlName="phoneNo"
          type="text"
          [placeholder]="'BOOKING.BOOKING_FORM.PHONE_NUMBER' | translate"
        />
        <ds-form-label>{{ 'BOOKING.BOOKING_FORM.PHONE_NUMBER' | translate }}</ds-form-label>
      </ds-input-wrapper>
      <ds-form-error *ngIf="formControls.phoneNo.hasError('pattern')">
        <div class="pt-1">{{ 'BOOKING.BOOKING_FORM.INLCUDE_COUNTRY_CODE_TEXT' | translate }}</div>
      </ds-form-error>
    </ng-container>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.phoneNo?.hasError('required') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.PHONE_NUMBER_REQUIRED' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.phoneNo?.hasError('pattern') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.PHONE_NUMBER_NOT_IN_VALID_FORMAT' | translate
      }}</ds-form-error>
    </div>
    <ds-textarea class="mt-4 text-left">
      <textarea
        dsInput
        formControlName="note"
        class="h-[150px]"
        [placeholder]="'BOOKING.BOOKING_FORM.SHARE_TEXT' | translate"
      ></textarea>
      <ds-form-label>{{ 'BOOKING.BOOKING_FORM.NOTES' | translate }}</ds-form-label>
    </ds-textarea>
    <ds-form-sub-text *ngIf="leftCountComments$ | async as leftCountComments" class="text-left">
      {{ 'BOOKING.BOOKING_FORM.COMMENTS_DESCRIPTION' | translate: { value: leftCountComments } }}
    </ds-form-sub-text>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.note?.hasError('maxlength') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.COMMENTS_MAX_LENGTH' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.note?.hasError('whitespace') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.ENTER_VALID_TEXT' | translate
      }}</ds-form-error>
    </div>
    <div class="mt-5" *ngIf="isMobileView$ | async">
      <button
        dsButton="default"
        class="w-full h-10 py-2 px-4 shadow-shadow-z200 text-center"
        [disabled]="(saveBookingPending$ | async)!"
        (click)="prevStep()"
      >
        {{ 'BOOKING.BOOKING_OTHER.GO_BACK' | translate }}
      </button>
    </div>
    <div class="mt-5">
      <button
        class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
        type="submit"
        [disabled]="(saveBookingPending$ | async)!"
      >
        {{ 'BOOKING.BOOKING_OTHER.CONFIRM_BOOKING' | translate }}
      </button>
    </div>
    <div class="w-full text-center mt-5">
      <ds-form-error *ngIf="error$ | async as errorMessage">{{ errorMessage }}</ds-form-error>
    </div>
  </form>
  <div class="flex flex-row justify-center">
    <ds-spinner *ngIf="saveBookingPending$ | async" [color]="spinnerColor" [size]="30"></ds-spinner>
  </div>
</div>
