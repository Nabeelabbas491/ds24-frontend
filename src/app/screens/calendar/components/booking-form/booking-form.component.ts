import { Component } from '@angular/core';
import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
import { ReactiveLifecycles } from '@ds24/utilities';
import { Store } from '@ngrx/store';
import { actions, selectors } from '../../../../store';
import { BehaviorSubject, filter, takeUntil } from 'rxjs';
import { MeetingTypeId } from './../../../../types/booking.type';
import { IconColor } from '@ds24/elements';
import { noWhitespaceValidator } from './../../../../shared/common/util';

@Component({
  selector: 'ds-booking-form',
  templateUrl: './booking-form.component.html',
})
export class BookingFormComponent extends ReactiveLifecycles {
  meetingTypeIds = MeetingTypeId;

  meetingType$ = this.store.select(selectors.booking.selectMeetingType);
  isMobileView$ = this.store.select(selectors.viewBreakPoint.selectIsMobileView);
  error$ = this.store.select(selectors.bookingAPI.selectBookingSaveError);
  saveBookingPending$ = this.store.select(selectors.bookingAPI.selectBookingSavePending);

  leftCountName$ = new BehaviorSubject<number>(80);
  leftCountComments$ = new BehaviorSubject<number>(300);

  spinnerColor = IconColor.Primary300;

  form: FormGroup = this.fb.group({
    name: new FormControl('', { validators: [Validators.required, Validators.maxLength(80), noWhitespaceValidator] }),
    email: new FormControl('', { validators: [Validators.required, Validators.maxLength(255)] }),
    phoneNo: new FormControl('', { validators: [Validators.pattern(/^\+\d{1,4}\s?\d{1,}$/)] }),
    note: new FormControl('', { validators: [Validators.maxLength(300), noWhitespaceValidator] }),
  });

  constructor(
    private fb: FormBuilder,
    private store: Store,
  ) {
    super();
  }

  ngOnInit() {
    super.ngOnInit();

    this.saveBookingPending$.pipe(takeUntil(this.ngOnDestroy$)).subscribe(isPending => {
      if (isPending) {
        this.form.disable();
      } else {
        this.form.enable();
      }
    });

    this.form
      .get('name')
      ?.valueChanges.pipe(takeUntil(this.ngOnDestroy$))
      .subscribe(value => {
        value ? this.leftCountName$.next(Math.max(0, 80 - value?.length)) : this.leftCountName$.next(80);
      });

    this.form
      .get('note')
      ?.valueChanges.pipe(takeUntil(this.ngOnDestroy$))
      .subscribe(value => {
        value ? this.leftCountComments$.next(Math.max(0, 300 - value?.length)) : this.leftCountComments$.next(300);
      });

    this.meetingType$
      .pipe(
        filter(meetingType => meetingType?.id === this.meetingTypeIds.Outbound),
        takeUntil(this.ngOnDestroy$),
      )
      .subscribe(() => {
        //eslint-disable-line
        this.formControls.phoneNo?.setValidators([Validators.required, Validators.pattern(/^\+\d{1,4}\s?\d{1,}$/)]);
      });
  }

  prevStep() {
    this.store.dispatch(actions.booking.selectStep({ step: 'booking_type' }));
  }
  confirmBooking() {
    this.form.markAllAsTouched();
    if (this.form.valid) {
      this.store.dispatch(actions.bookingAPI.saveBooking({ bookingDetail: this.form.value }));
    }
  }

  get formControls(): any {
    return this.form.controls;
  }
}
