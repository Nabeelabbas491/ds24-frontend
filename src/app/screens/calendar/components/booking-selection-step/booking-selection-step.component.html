<div class="flex flex-row justify-start bg-primary">
  <div class="flex flex-col justify-start items-center w-[56%] bg-neutral-n50">
    <ds-booking-user-profile [bookingTemplateSummary]="bookingTemplateSummary$ | async"></ds-booking-user-profile>
    <h4 class="font-medium my-3">{{ 'CALENDAR.SELECT_A_DATE' | translate }}</h4>
    <ds-calendar></ds-calendar>
  </div>
  <div class="w-[44%] bg-neutral-n0 p-5" *ngIf="selectedDay$ | async">
    <ds-booking-type-selection
      [meetingType]="meetingType$ | async"
      (meetingTypeChange)="meetingTypeChange($event)"
    ></ds-booking-type-selection>
    <div class="mt-8">
      <ds-time-selection></ds-time-selection>
    </div>
    <div class="mt-5" *ngIf="isBookingSelectionStepComplete$ | async">
      <button
        class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
        (click)="nextStep()"
      >
        {{ 'BOOKING.BOOKING_OTHER.NEXT' | translate }}
      </button>
    </div>
  </div>
</div>
