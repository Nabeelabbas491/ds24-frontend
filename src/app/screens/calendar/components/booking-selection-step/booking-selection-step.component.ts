import { Component } from '@angular/core';
import { Store } from '@ngrx/store';
import { actions, selectors } from '../../../../store';
import { MeetingType } from 'src/app/types/misc.type';

@Component({
  selector: 'ds-booking-selection-step',
  templateUrl: './booking-selection-step.component.html',
})
export class BookingSelectionStepComponent {
  selectedDay$ = this.store.select(selectors.booking.selectSelectedDay);
  isBookingSelectionStepComplete$ = this.store.select(selectors.booking.isBookingSelectionStepComplete);
  meetingType$ = this.store.select(selectors.booking.selectMeetingType);
  bookingTemplateSummary$ = this.store.select(selectors.booking.selectBookingTemplateSummary);

  constructor(private store: Store) {}
  meetingTypeChange(meetingType: MeetingType) {
    this.store.dispatch(actions.booking.meetingType({ meetingType }));
  }

  nextStep() {
    this.store.dispatch(actions.booking.selectStep({ step: 'booking_form' }));
  }
}
