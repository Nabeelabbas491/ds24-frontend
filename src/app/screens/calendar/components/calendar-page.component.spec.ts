import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TranslateModule } from '@ngx-translate/core';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { State, selectors } from './../../../store';
import { MemoizedSelector } from '@ngrx/store';
import { CalendarPageComponent } from './calendar-page.component';
import { BookingTypeSelectionComponent } from './booking-type-selection/booking-type-selection.component';
import { TimeSelectionComponent } from './time-selection/time-selection.component';
import { BookingFormComponent } from './booking-form/booking-form.component';
import { BookingSummaryComponent } from './booking-summary/booking-summary.component';
import { BookingStep } from './../../../types/booking.type';
import { DateFormatPipe } from './../../../shared/pipes/date-format.pipe';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BookingConfirmationComponent } from './booking-confirmation/booking-confirmation.component';
import { BookingSelectionStepComponent } from './booking-selection-step/booking-selection-step.component';
import { BookingSummaryStepComponent } from './booking-summary-step/booking-summary-step.component';
import { BookingConfirmationStepComponent } from './booking-confirmation-step/booking-confirmation-step.component';
import { TimeSlotPipe } from './../../../shared/pipes/time-slot.pipe';

describe('Step1Component', () => {
  let fixture: ComponentFixture<CalendarPageComponent>;
  let instance: CalendarPageComponent;
  let mockStore: MockStore<State>;

  let currentStep: MemoizedSelector<State, BookingStep>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        CalendarPageComponent,
        BookingSelectionStepComponent,
        BookingTypeSelectionComponent,
        TimeSelectionComponent,
        BookingSummaryStepComponent,
        BookingFormComponent,
        BookingSummaryComponent,
        BookingConfirmationStepComponent,
        BookingConfirmationComponent,
        DateFormatPipe,
        TimeSlotPipe,
      ],
      imports: [TranslateModule.forRoot(), FormsModule, ReactiveFormsModule],
      providers: [provideMockStore()],
    });
    fixture = TestBed.createComponent(CalendarPageComponent);
    instance = fixture.componentInstance;
    mockStore = TestBed.inject(MockStore);

    currentStep = mockStore.overrideSelector(selectors.booking.selectCurrentStep, 'calendar');

    fixture.detectChanges();

    jest.spyOn(mockStore, 'dispatch');
  });

  it('should compile', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should demonstrate bookingType change', () => {
    currentStep.setResult('booking_form');
    mockStore.refreshState();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should create', () => {
    expect(instance).toBeTruthy();
  });
});
