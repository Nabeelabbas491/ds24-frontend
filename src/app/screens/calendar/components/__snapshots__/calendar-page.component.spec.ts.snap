// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step1Component should compile 1`] = `
<ds-calendar-page
  currentStep$={[Function _Store]}
  isMobileView$={[Function _Store]}
  messageTypeError={[Function Number]}
  selectBookingTemplateError$={[Function _Store]}
  selectBookingTemplatePending$={[Function _Store]}
  spinnerColor={[Function String]}
  store={[Function _MockStore]}
>
  <div
    class="border-neutral-n200 shadow-shadow-z100 max-w-4xl my-4 mx-auto rounded-lg overflow-hidden"
  >
    <ds-booking-selection-step>
      <div
        class="flex flex-row justify-start bg-primary"
      >
        <div
          class="flex flex-col justify-start items-center w-[56%] bg-neutral-n50"
        >
          <ds-booking-user-profile />
          <h4
            class="font-medium my-3"
          >
            CALENDAR.SELECT_A_DATE
          </h4>
          <ds-calendar />
        </div>
      </div>
    </ds-booking-selection-step>
  </div>
</ds-calendar-page>
`;

exports[`Step1Component should demonstrate bookingType change 1`] = `
<ds-calendar-page
  currentStep$={[Function _Store]}
  isMobileView$={[Function _Store]}
  messageTypeError={[Function Number]}
  selectBookingTemplateError$={[Function _Store]}
  selectBookingTemplatePending$={[Function _Store]}
  spinnerColor={[Function String]}
  store={[Function _MockStore]}
>
  <div
    class="border-neutral-n200 shadow-shadow-z100 max-w-4xl my-4 mx-auto rounded-lg overflow-hidden"
  >
    <ds-booking-summary-step>
      <div
        class="flex flex-row flex-wrap justify-start bg-primary"
      >
        <div
          class="flex flex-col justify-start items-center mobile-view:items-start mobile-view:ml-5 w-[56%] mobile-view:w-full"
        >
          <ds-booking-summary>
            <ds-booking-info />
            <ds-booking-info />
            <ds-booking-info />
          </ds-booking-summary>
        </div>
        <div
          class="w-[44%] mobile-view:w-full mobile-view:mt-5 bg-neutral-n0 p-5"
        >
          <ds-booking-form>
            <div>
              <h4
                class="font-medium"
              >
                BOOKING.BOOKING_FORM.ENTER_DETAILS
              </h4>
              <form
                class="ng-untouched ng-pristine ng-invalid"
                novalidate=""
              >
                <ds-input-wrapper
                  class="mt-2 text-left"
                >
                  <input
                    class="ng-untouched ng-pristine ng-invalid"
                    dsinput=""
                    formcontrolname="name"
                    placeholder="BOOKING.BOOKING_FORM.NAME"
                    type="text"
                  />
                  <ds-form-label>
                    BOOKING.BOOKING_FORM.NAME
                  </ds-form-label>
                </ds-input-wrapper>
                <ds-form-sub-text
                  class="text-left"
                >
                   BOOKING.BOOKING_FORM.NAME_DESCRIPTION 
                </ds-form-sub-text>
                <div
                  class="w-full text-left"
                >
                </div>
                <ds-input-wrapper
                  class="mt-2 text-left"
                >
                  <input
                    class="ng-untouched ng-pristine ng-invalid"
                    dsinput=""
                    email=""
                    formcontrolname="email"
                    placeholder="BOOKING.BOOKING_FORM.EMAIL"
                    type="email"
                  />
                  <ds-form-label>
                    BOOKING.BOOKING_FORM.EMAIL
                  </ds-form-label>
                </ds-input-wrapper>
                <div
                  class="w-full text-left"
                >
                </div>
                <div
                  class="w-full text-left"
                >
                </div>
                <ds-textarea
                  class="mt-4 text-left"
                >
                  <textarea
                    class="h-[150px] ng-untouched ng-pristine ng-valid"
                    dsinput=""
                    formcontrolname="note"
                    placeholder="BOOKING.BOOKING_FORM.SHARE_TEXT"
                  />
                  <ds-form-label>
                    BOOKING.BOOKING_FORM.NOTES
                  </ds-form-label>
                </ds-textarea>
                <ds-form-sub-text
                  class="text-left"
                >
                   BOOKING.BOOKING_FORM.COMMENTS_DESCRIPTION 
                </ds-form-sub-text>
                <div
                  class="w-full text-left"
                >
                </div>
                <div
                  class="mt-5"
                >
                  <button
                    class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
                    type="submit"
                  >
                     BOOKING.BOOKING_OTHER.CONFIRM_BOOKING 
                  </button>
                </div>
                <div
                  class="w-full text-center mt-5"
                >
                </div>
              </form>
              <div
                class="flex flex-row justify-center"
              >
              </div>
            </div>
          </ds-booking-form>
        </div>
      </div>
    </ds-booking-summary-step>
  </div>
</ds-calendar-page>
`;
